addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -3188,7 +3188,24 @@ export class OdooEditor extends EventTarget {
         this.deleteRange();
     }
     _onClipboardCopy(clipboardEvent) {
-        if (this.isSelectionInEditable()) {
+        const hasPartiallySelectedTable = hasTableSelection(this.editable) &&
+            ![...this.editable.querySelectorAll('.o_selected_table td')].every(td => (
+                !closestElement(td, 'table').classList.contains('o_selected_table') || // ignore nested
+                td.classList.contains('o_selected_td')
+            ));
+        if (hasPartiallySelectedTable) {
+            clipboardEvent.preventDefault();
+            const table = this.editable.querySelector('.o_selected_table');
+            const copyTable = table.cloneNode();
+            for (const tr of table.querySelectorAll('tr:has(.o_selected_td)')) {
+                const copyTr = tr.cloneNode();
+                for (const td of tr.querySelectorAll('.o_selected_td')) {
+                    copyTr.append(td.cloneNode(true));
+                }
+                copyTable.append(copyTr);
+            }
```

Comment:
Just a suggestion, maybe that wouldn't work or that wouldn't be better, it's hard to say from the diff here.

Why not `cloneNode(true)` the entire table then `querySelectorAll` the `td` which do _not_ have `.o_selected_td`, call `.remove()` on them then remove the now-empty tr if any ?

Reply:
It works fine but it's not necessary simpler. I made a commit with a version doing that, which works with nested tables.